### PR TITLE
Add events to calendar

### LINKED ISSUE #37: Add Events to Calendar

### PR DESCRIPTION
**Pull request by Jdellis818**
This PR adds events to the calendars and allows you to scroll through semesters. It does so by creating a `GroupedEvents` objects which groups events by the faculty member and then loads the events on their calendar.

### CODE REVIEW

File: client-course-schedulizer/src/components/Toolbar/SemesterSelector/SemesterSelector.tsx
Comment (charkour) [inline client-course-schedulizer/src/components/Toolbar/SemesterSelector/SemesterSelector.tsx]: This onClick handler I would make into a function take takes an index parameter since it is used here and below. If we update the logic in one, it'll need to be updated in both.

File: client-course-schedulizer/src/components/Toolbar/SemesterSelector/SemesterSelector.tsx
Comment (charkour) [inline client-course-schedulizer/src/components/Toolbar/SemesterSelector/SemesterSelector.tsx]: Wow, I didn't know this was possible. Iterating over enumerations in typescript is kinda weird. haha nice job

File: client-course-schedulizer/src/utilities/services/schedule.ts
Comment (charkour) [inline client-course-schedulizer/src/utilities/services/schedule.ts]: I've seen this a couple places. Would it be better to make a function to do this? 

```
const enumArray = <T>(e: T) => {
  return Object.keys(e).map((val) => {
    return e[val as keyof typeof e];
  });
};
```

Comment (charkour) [reply client-course-schedulizer/src/utilities/services/schedule.ts]: use: `const days: Day[] = enumArray(Day);`

File: client-course-schedulizer/src/components/reuseables/Schedule/Schedule.tsx
Comment (charkour) [inline client-course-schedulizer/src/components/reuseables/Schedule/Schedule.tsx:37]: I would just say `// add context times to the existing calendarOptions`

File: client-course-schedulizer/src/components/reuseables/Schedule/Schedule.tsx
Comment (charkour) [inline client-course-schedulizer/src/components/reuseables/Schedule/Schedule.tsx]: It might be worth doing research to see if it is important that we move logic out of the function component. I think these get redefined on every render, but maybe we put these in a useCallback()? Not sure. would be worth researching.

**Review by charkour — CHANGES_REQUESTED** [review]
This is some great work! I like how seemlessly it works together and seeing the calendars with the events is awesome. I like to use of moment and also how you wrote code that is extensible and flexible. I think it would be worth to add some unit tests in the future to test the sorting and filtering logic. 

A couple small things to address before merging.

File: client-course-schedulizer/src/components/reuseables/Schedule/Schedule.tsx
Comment (Jdellis818) [reply client-course-schedulizer/src/components/reuseables/Schedule/Schedule.tsx]: I changed this to `useMemo`, but I'm unsure whether the dependencies do anything since they include objects and I'm worried that it does a reference comparison rather than a value comparison. If it uses reference comparison then I think the code would re-run on any re-render since an object would have a new location in memory even if it had the same values.

Comment (Jdellis818) [reply client-course-schedulizer/src/components/reuseables/Schedule/Schedule.tsx]: I did some research though and it seems like it may not be worth worrying about anyway since the operation is not computationally expensive. React is fast enough that a re-render isn't a big deal

**Review by charkour — APPROVED** [review]
I did some preliminary research on `useMemo()` in our app and counted the number of re-renderes. When `useMemo()` is not used, `forOwn()` is called 15 times and `calendarHeadersNoEmptyInTerm()` is called 15 times as well; however, with the memoization, `filteredEvents` and `calendarHeadersNoEmptyInTerm` are only redefined twice each. I think getting these called twice is fine! In my experience, it is really hard to get things called twice by hooks to be called only once since there are many complexities that I do not understand. 

I liked the rest of the updates. You can merge this when ready!

**Pull request merged**
